dali/base/dafdesc.cpp

```diff
@@ -3156,46 +3316,25 @@ void initializeStorageGroups(bool createPlanesFromGroups)
 #ifndef _CONTAINERIZED
     if (createPlanesFromGroups && !storage->hasProp("planes"))
     {
+        GroupInfoArray allGroups;
+
         //Create information about the storage planes from the groups published in dali
-        INamedGroupStore & groupStore = queryNamedGroupStore();
-        Owned<INamedGroupIterator> iter= groupStore.getIterator();
-        ForEach(*iter)
+        //Use the Groups section directly, rather than queryNamedGroupStore(), so that hostnames are preserved
         {
-            StringBuffer name, dir;
-            iter->get(name);
-            if (name.length())
+            Owned<IRemoteConnection> conn = querySDS().connect("/Groups", myProcessSession(), RTM_LOCK_READ, SDS_CONNECT_TIMEOUT);
+            Owned<IPropertyTreeIterator> groups = conn->queryRoot()->getElements("Group");
+            ForEach(*groups)
             {
-                IPropertyTree * plane = storage->addPropTree("planes");
-                plane->setProp("@name", name);
-                GroupType groupType;
-                Owned<IGroup> group = groupStore.lookup(name, dir, groupType);
-
-                if (dir.length())
-                    plane->setProp("@prefix", dir);
-                else
-                    plane->setProp("@prefix", queryBaseDirectory(groupType, 0));
-
-                if (!group)
-                    plane->setPropInt("@numDevices", 0);
-                else if (group->ordinality() != 1)
-                    plane->setPropInt("@numDevices", group->ordinality());
-
-                if (groupType == grp_thor)
+                IPropertyTree & group = groups->query();
+                GroupInformation * next = new GroupInformation(group.queryProp("@name"));
```

Comment:
perhaps make this an Owned and use getClear where ownership transferred, just in case interim leads to leak